modules/katamari/src/test/ts/atomic/api/data/ResultCovarianceAssertionTest.ts

```diff
@@ -0,0 +1,29 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { assert } from 'chai';
+
+import { Result } from 'ephox/katamari/api/Result';
+import * as Results from 'ephox/katamari/api/Results';
+
+interface Animal {
+  readonly name: string;
+}
+
+interface Cat extends Animal {
+  readonly length: number;
+}
+
+// https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)
+//
+// We need to make sure that Result<T, E> is, and remains, covariant with
+// respect to T and E.
+describe('atomic.katamari.api.data.ResultCovarianceAssertionTest', () => {
```

Comment:
Might as well merge the pair of these into a single `CovarianceAssertionTest` and get Optional and Result in the same group — 99% of the code is the same.

Also, does this test still pass with strict mode on (I know everything else in katamari breaks, just want to make sure this file in particular doesn't have any errors so we don't get bitten like last time)

Reply:
Have merged the 2 tests into a single file. Also yes, I double checked and there's no errors in this test when strict mode is enabled.